Tools/deploy-mac.py

```diff
@@ -0,0 +1,172 @@
+	parent, child = findFramework(splitPath(path))
+	return joinPath(parent[:-2])
+
+def makedirs(path):
+	split = splitPath(path)
+	accum = []
+	split.reverse()
+	while split:
+		accum.append(split.pop())
+		newPath = joinPath(accum)
+		if newPath == '/':
+			continue
+		try:
+			os.mkdir(newPath)
+		except OSError as e:
+			if e.errno != errno.EEXIST:
+				raise
+
+
+def parseOtoolLine(line, execPath, root):
+	if not line.startswith('\t'):
+		return None, None, None, None
+	line = line[1:]
+	match = re.match('([@/].*) \(compatibility version.*\)', line)
+	path = match.group(1)
+	split = splitPath(path)
+	newExecPath = ['@executable_path', '..', 'Frameworks']
+	newPath = execPath[:-1]
+	newPath.append('Frameworks')
+	if split[:3] == ['/', 'usr', 'lib'] or split[:2] == ['/', 'System']:
+		return None, None, None, None
+	if split[0] == '@executable_path':
+		split[:1] = execPath
+	if split[0] == '/' and not os.access(joinPath(split), os.F_OK):
+		split[:1] = root
+	try:
+		oldPath = joinPath(split)
+		while True:
+			linkPath = os.readlink(os.path.abspath(oldPath))
+			oldPath = os.path.join(os.path.dirname(oldPath), linkPath)
+	except OSError as e:
+		if e.errno != errno.EINVAL:
+			raise
+		split = splitPath(oldPath)
+	isFramework = False
+	if not split[-1].endswith('.dylib'):
+		isFramework = True
+		split, framework = findFramework(split)
+	newPath.append(split[-1])
+	newExecPath.append(split[-1])
+	if isFramework:
+		newPath.extend(framework)
+		newExecPath.extend(framework)
+		split.extend(framework)
+	newPath = joinPath(newPath)
+	newExecPath = joinPath(newExecPath)
+	return joinPath(split), newPath, path, newExecPath
+
+def updateMachO(bin, execPath, root):
+	global qtPath
+	otoolOutput = subprocess.check_output([otool, '-L', bin])
+	toUpdate = []
+	for line in otoolOutput.split('\n'):
+		oldPath, newPath, oldExecPath, newExecPath = parseOtoolLine(line, execPath, root)
+		if not newPath:
+			continue
+		if os.access(newPath, os.F_OK):
+			if verbose:
+				print('Skipping copying {}, already done.'.format(oldPath))
+		elif os.path.abspath(oldPath) != os.path.abspath(newPath):
+			if verbose:
+				print('Copying {} to {}...'.format(oldPath, newPath))
+			parent, child = os.path.split(newPath)
+			makedirs(parent)
+			shutil.copy2(oldPath, newPath)
+			os.chmod(newPath, 0o644)
+		toUpdate.append((newPath, oldExecPath, newExecPath))
+		if not qtPath and 'Qt' in oldPath:
+			qtPath = findQtPath(oldPath)
+			if verbose:
+				print('Found Qt path at {}.'.format(qtPath))
+	for path, oldExecPath, newExecPath in toUpdate:
+		if path != bin:
+			updateMachO(path, execPath, root)
+			if verbose:
+				print('Updating Mach-O load from {} to {}...'.format(oldExecPath, newExecPath))
+			subprocess.check_call([installNameTool, '-change', oldExecPath, newExecPath, bin])
+		else:
+			if verbose:
+				print('Updating Mach-O id from {} to {}...'.format(oldExecPath, newExecPath))
+			subprocess.check_call([installNameTool, '-id', newExecPath, bin])
```

Comment:
Huh? I didn't think install_name_tool could take multiple inputs at once. Furthermore, this function is recursive, so collapsing all of those...I'm not sure if that would have the intended effect.